src/server/game/Entities/Player/PlayerUpdates.cpp

```diff
@@ -0,0 +1,2311 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU
+ * GPL v2 license, you may redistribute it and/or modify it under version 2 of
+ * the License, or (at your option), any later version. Copyright (C) 2008-2016
+ * TrinityCore <http://www.trinitycore.org/> Copyright (C) 2005-2009 MaNGOS
+ * <http://getmangos.com/>
+ */
+
+#include "BattlefieldMgr.h"
+#include "CellImpl.h"
+#include "Channel.h"
+#include "ChannelMgr.h"
+#include "Formulas.h"
+#include "GridNotifiers.h"
+#include "GridNotifiersImpl.h"
+#include "Group.h"
+#include "Guild.h"
+#include "InstanceScript.h"
+#include "Language.h"
+#include "OutdoorPvPMgr.h"
+#include "Pet.h"
+#include "Player.h"
+#include "SavingSystem.h"
+#include "ScriptMgr.h"
+#include "SkillDiscovery.h"
+#include "SpellAuraEffects.h"
+#include "UpdateFieldFlags.h"
+#include "Vehicle.h"
+#include "WeatherMgr.h"
+
+// Zone Interval should be 1 second
+#define ZONE_UPDATE_INTERVAL (1 * IN_MILLISECONDS)
+
+#define CINEMATIC_UPDATEDIFF 500
+#define CINEMATIC_LOOKAHEAD  (2 * IN_MILLISECONDS)
+
+void Player::Update(uint32 p_time)
+{
+  if(!IsInWorld())
+    return;
+
+  sScriptMgr->OnBeforePlayerUpdate(this, p_time);
+
+  // undelivered mail
+  if(m_nextMailDelivereTime && m_nextMailDelivereTime <= time(nullptr))
```

Comment:
```suggestion
  if (m_nextMailDelivereTime && m_nextMailDelivereTime <= time(nullptr))
```

Reply:
I will make changed to https://github.com/azerothcore/azerothcore-wotlk/pull/6848 to fix this.